cardano-crypto-class/src/Cardano/Crypto/Libsodium/Constants.hsc

```diff
@@ -0,0 +1,24 @@
+{-# LANGUAGE DataKinds #-}
+module Cardano.Crypto.Libsodium.Constants (
+    CRYPTO_SHA256_BYTES,
+    CRYPTO_SHA512_BYTES,
+    CRYPTO_BLAKE2B_256_BYTES,
+    CRYPTO_SHA256_STATE_SIZE,
+    CRYPTO_SHA512_STATE_SIZE,
+    CRYPTO_BLAKE2B_256_STATE_SIZE,
+    )  where
+
+#include <sodium.h>
+
+-- From https://libsodium.gitbook.io/doc/advanced/sha-2_hash_function
+-- and https://libsodium.gitbook.io/doc/hashing/generic_hashing
+
+type CRYPTO_SHA256_BYTES = #{const crypto_hash_sha256_BYTES}
```

Comment:
Could use pattern synonyms for these, and then we can match on them

Reply:
Note: `type`, not value.

Reply:
Yep, sorry, saw how you were using it below.